app/dao/inbound_sms_dao.py

```diff
@@ -93,3 +78,48 @@ def dao_get_inbound_sms_by_id(service_id, inbound_id):
         id=inbound_id,
         service_id=service_id
     ).one()
+
+
+def dao_get_paginated_most_recent_inbound_sms_by_user_number_for_service(
+    service_id,
+    page
+):
+    """
+    This query starts from inbound_sms and joins on to itself to find the most recent row for each user_number
```

Comment:
I'm happy that this query is doing the right thing. I confirmed the results by comparing it with this query:
SELECT distinct(user_number), notify_number, max(created_at)                                                                                                     from inbound_sms                                                                                                                                                                                                  
WHERE service_id = :service_id 
GROUP BY user_number, notify_number 
ORDER BY max(created_at) desc
LIMIT :page_size OFFSET :page